src/handler.ts

```diff
@@ -242,7 +286,17 @@ abstract class AbstractFileHandler implements IMessageFilter {
     }
   }
 
-  private _manager: IContentsManager;
+  /**
+   * Handle a focus events.
+   */
+  private _onFocus(event: Event) {
+    let target = event.target as HTMLElement;
+    let widget = arrays.find(this._widgets,
+      w => w.isVisible && w.node.contains(target));
+    if (widget) AbstractFileHandler.activeWidgetProperty.set(this, widget);
```

Comment:
This should emit an `activated` signal if was previous not active (meaning *no* widgets focused), but now is (meaning *some* widget focued). So changing focus from one text editor to another, should *not* emit the `activated` signal of the text file handler (if I'm following things correctly).